Quorum/ipfs_validator.py

```diff
@@ -0,0 +1,69 @@
+from Quorum.utils.chain_enum import Chain
+import Quorum.config as config
+from Quorum.apis.block_explorers.chains_api import ChainAPI
+from Quorum.apis.llms.claude_conversation import ClaudeConversation
+
+import requests
+import argparse
+import re
+from pathlib import Path
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description='Compare ipfs with actual payload.')
+    parser.add_argument('--proposal_id', type=int, help='The id of the proposal.')
+    parser.add_argument('--chain', type=str, choices=[chain.value for chain in Chain], help='Blockchain chain.')
+    parser.add_argument('--proposal_address', type=str, help='Ethereum proposal address.')
+    parser.add_argument('--prompt_template', type=Path, default=Path(__file__).parent / 'ipfs_validation_prompt.txt',
+                        help='Optional other template for prompting the LLM.')
+    
+    args = parser.parse_args()
+    if not args.prompt_template.exists():
+        raise argparse.ArgumentError(f'Could not find prompts at {args.prompt_template}.')
+
+    return args
+
+
+def get_raw_ipfs(proposal_id: int) -> str:
```

Comment:
I think we should have a cache machnisem here to save both time and money.
If you use the same proposal and the same prompts there is no need to run the entire process every time.

Reply:
I didn't really want to do it since this function is temporary but I implemented something very crude.